include/pybind11/cast.h

```diff
@@ -31,6 +31,8 @@
 PYBIND11_NAMESPACE_BEGIN(PYBIND11_NAMESPACE)
 PYBIND11_NAMESPACE_BEGIN(detail)
 
+PYBIND11_DISABLE_WARNING_MSVC(4127)
```

Comment:
Do we not bother popping this? As annoying as it is, it is somewhat useful in knowing what should be constexpr in C++17 codebases. I think we should disable within pybind11 namescope, but not for the end user (unless there is no way to do this for this specific pragma?)

Reply:
This only disables the warning within the detail namespace.

There is a warning push/pop at every PYBIND11_NAMESPACE_BEGIN and PYBIND11_NAMESPACE_END

So this warning will only be disabled until PYBIND11_NAMESPACE_END(detail)